packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/table/PaginationUtil.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2024 CEA LIST.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.papaya.representations.table;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.sirius.components.collaborative.tables.CursorBasedPaginationData;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.components.tables.descriptions.PaginatedData;
+
+/**
+ * Util class to help to paginate lists.
+ *
+ * @author frouene
+ */
+public class PaginationUtil {
+
+    public PaginatedData getPaginatedData(CursorBasedPaginationData pagination, List<Object> allRows, IEditingContext editingContext, IObjectService objectService) {
```

Comment:
You could remove the whole class unless I'm mistaken because you are starting by considering that you have access to everything an then you are just removing what you don't need. This is explicitly what I don't want. I want to compute only what I need. This PR should **prove** that we can remove calls to eAllContents.